contracts/RootChain.sol

```diff
@@ -326,62 +331,88 @@ contract RootChain {
         require(_transactionIncluded(_outputTx, _outputId, _outputTxInclusionProof));
 
         // Decode the output ID.
-        uint256 oindex = _outputId.getOindex();
+        uint8 oindex = uint8(_outputId.getOindex());
 
         // Parse outputTx.
         PlasmaCore.TransactionOutput memory output = _outputTx.getOutput(oindex);
 
         // Only output owner can start an exit.
         require(msg.sender == output.owner);
 
-        uint192 exitId = getStandardExitId(_outputId);
+        uint192 exitId = getStandardExitId(keccak256(_outputTx), oindex);
 
         // Make sure this exit is valid.
         require(output.amount > 0);
         require(exits[exitId].amount == 0);
 
+        InFlightExit storage inFlightExit = _getInFlightExit(_outputTx);
+        if (inFlightExit.exitStartTimestamp != 0) {
+            // Check if this output was piggybacked or exited in in-flight exit
+            require(!inFlightExit.exitMap.bitSet(oindex + MAX_INPUTS) && !inFlightExit.exitMap.bitSet(oindex + MAX_INPUTS + MAX_INPUTS*2));
```

Comment:
let's put all the `bitSet` (check) and setting those bits into domain-named private functions, that explain what is being flagged / checked from the perspective of the contract logic.